pkg/idp/idp.go

```diff
@@ -196,6 +207,33 @@ func (s *Service) DeleteProviderApps(appName string, ctx context.Context) error
 	return nil
 }
 
+func (s *Service) secretNeedsUpdate(oldConnectors map[string]dex.Connector, newConnectors map[string]dex.Connector) bool {
```

Comment:
Here we compare old connector configuration to new connector configuration. (see comments)